### PR TITLE
Add support for UDP on IPv6 "unconnected" sends.

### PR DESCRIPTION
**Pull request by brycekahle**
### What does this PR do?

Similar to #6307 but for IPv6 UDP sends, this adds the ability to extract addresses and ports for UDP sends that are "unconnected" (meaning they haven't received a reply yet).

### Motivation

Improving IPv6 and UDP coverage of network tracer.

### Additional Notes

The 7th and later arguments to a function on `x86_64` are stored on the stack, which is why the macro for `PT_REGS_PARM7` is so complicated. See https://eli.thegreenplace.net/2011/09/06/stack-frame-layout-on-x86-64 for more information.

I was unable to use a public IPv6 destination address for offset guessing because the agent dev VM does not have a default IPv6 route. This is likely a common setup, so I worked around it by using link-local and multicast addresses.

### Describe your test plan

Added test comparable to IPv4 version. It passes in the agent VM.

### CODE REVIEW

**Review by hmahmood — APPROVED** [review]

**Comment by brycekahle** [comment]
Moving to DRAFT. This is complicated by the fact that the `flowi6` parameter shifts from the 9th argument to the 7th at kernel 4.7. This is easily handled by runtime compilation, but not prebuilt binaries.

**Comment by brycekahle** [comment]
Found a solution for both prebuilt and runtime compiled versions.

File: pkg/network/ebpf/c/prebuilt/offset-guess.c [hunk 1]
```diff
@@ -145,7 +158,41 @@ int kprobe__ip_make_skb(struct pt_regs* ctx) {
...
+SEC("kprobe/ip6_make_skb")
+int kprobe__ip6_make_skb(struct pt_regs* ctx) {
+    u64 zero = 0;
+    tracer_status_t* status = bpf_map_lookup_elem(&tracer_status, &zero);
+
+    if (status == NULL) {
+        return 0;
+    }
+    struct flowi6* fl6 = (struct flowi6*)PT_REGS_PARM7(ctx);
+    if (fl6 == NULL) {
```
Comment (hmahmood) [inline pkg/network/ebpf/c/prebuilt/offset-guess.c:174]: Why would this be `NULL`?

File: pkg/network/ebpf/c/runtime/tracer.c [hunk 2]
```diff
@@ -255,20 +248,63 @@ int kretprobe__tcp_close(struct pt_regs* ctx) {
 
 SEC("kprobe/ip6_make_skb")
 int kprobe__ip6_make_skb(struct pt_regs* ctx) {
+#ifdef FEATURE_IPV6_ENABLED
```
Comment (hmahmood) [inline pkg/network/ebpf/c/runtime/tracer.c:251]: Could we move this to before line 249 to completely compile this out if this flat is not set?

**Review by hmahmood — APPROVED** [review]

**Review by ogaca-dd — APPROVED** [review]
LGTM for `pkg/util/kernel/ip.go`.

File: pkg/network/ebpf/c/runtime/tracer.c (hunk 2 above)
Comment (brycekahle) [reply pkg/network/ebpf/c/runtime/tracer.c]: Turns out, yes! I've updated the code.

File: pkg/util/kernel/ip.go [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package kernel
```
Comment (sgnn7) [inline pkg/util/kernel/ip.go:1]: Just a drive-by comment but does this file need `// +build linux` preamble?

Comment (brycekahle) [reply pkg/util/kernel/ip.go:3]: Yes! Thanks for driving by. 😁

File: pkg/network/ebpf/c/prebuilt/offset-guess.c (hunk 1 above)
Comment (brycekahle) [reply pkg/network/ebpf/c/prebuilt/offset-guess.c]: Fixed.
